drivers/sensors/cirque_pinnacle.c

```diff
@@ -138,9 +140,10 @@ void ERA_ReadBytes(uint16_t address, uint8_t* data, uint16_t count) {
         RAP_Write(ERA_CONTROL, 0x05); // Signal ERA-read (auto-increment) to Pinnacle
 
         // Wait for status register 0x1E to clear
+        timeout_timer = timer_read();
         do {
             RAP_ReadBytes(ERA_CONTROL, &ERAControlValue, 1);
-        } while (ERAControlValue != 0x00);
+        } while ((ERAControlValue != 0x00) && (timer_elapsed(timeout_timer) <= CIRQUE_PINNACLE_TIMEOUT));
```

Comment:
At least on I2C we could use ` && touchpad_init==true` directly, as this will be set to false if there is any I2C error (or timeout). Not sure if SPI considers `CIRQUE_PINNACLE_TIMEOUT` at all?

Reply:
SPI doesn't have ack or check for any timeout. It will return whatever state MISO happens to be floating in, if the device is not connected. That could be either 0xFF or 0x00 (or something inbetween if there's noise on the line?).
Looking at AVR's _spi_master.c_ for example: `spi_start()` always returns true, except if chip select pin is invalid but that's an error before ever interacting with the device. So `touch_init` check can't really apply for SPI.

Reply:
This actually fixed a dead lock for me that happened with the code before. I started a small refactoring of the `RAP_Read/Write` functions. If you find it useful feel free to include it in this pr [cirque_timeout.patch.txt](https://github.com/qmk/qmk_firmware/files/9073188/cirque_timeout.patch.txt)

Reply:
As a side note, if a single I2C transfer fails the whole driver will turn into an invalid state - which requires a reboot of the keyboard. I can think of mechanism on the pointing device driver level that will try to re-init a driver in such a case after a certain timeout. But this is not in the scope of this PR.

Reply:
I think handling `touch_init` and I2C transaction failures adds too much to this PR, let's handle that in a separate one.